spring-integration-cloudevents/src/main/java/org/springframework/integration/cloudevents/transformer/ToCloudEventTransformer.java

```diff
@@ -0,0 +1,288 @@
+		if (!StringUtils.hasText(id)) {
+			throw new MessageTransformationException(message, "No id was found with the specified expression");
+		}
+
+		URI source = this.sourceExpression.getValue(this.evaluationContext, message, URI.class);
+		if (source == null) {
+			throw new MessageTransformationException(message, "No source was found with the specified expression");
+		}
+
+		String type = this.typeExpression.getValue(this.evaluationContext, message, String.class);
+		if (type == null) {
+			throw new MessageTransformationException(message, "No type was found with the specified expression");
+		}
+
+		String contentType = message.getHeaders().get(MessageHeaders.CONTENT_TYPE, String.class);
+		if (contentType == null) {
+			throw new MessageTransformationException(message, "Missing 'Content-Type' header");
+		}
+
+		EventFormat eventFormat = this.eventFormatProvider.resolveFormat(contentType);
+		if (eventFormat == null) {
+			throw new MessageTransformationException("No EventFormat found for '" + contentType + "'");
+		}
+
+		ToCloudEventTransformerExtensions extensions =
+				new ToCloudEventTransformerExtensions(this.evaluationContext, (Message<byte[]>) message,
+						this.cloudEventExtensionExpressions);
+
+		CloudEvent cloudEvent = CloudEventBuilder.v1()
+				.withId(id)
+				.withSource(source)
+				.withType(type)
+				.withTime(OffsetDateTime.now())
+				.withDataContentType(contentType)
+				.withDataSchema(this.dataSchemaExpression.getValue(this.evaluationContext, message, URI.class))
+				.withSubject(this.subjectExpression.getValue(this.evaluationContext, message, String.class))
+				.withData(getPayload(message))
+				.withExtension(extensions)
+				.build();
+
+		return MessageBuilder.withPayload(eventFormat.serialize(cloudEvent))
+				.copyHeaders(message.getHeaders())
+				.build();
+	}
+
+	@Override
+	public String getComponentType() {
+		return "ce:to-cloudevents-transformer";
+	}
+
+	private byte[] getPayload(Message<?> message) {
+		if (message.getPayload() instanceof byte[] messagePayload) {
+			return  messagePayload;
+		}
+		throw new MessageTransformationException("Message payload is not a byte array");
+	}
+
+	private static class ToCloudEventTransformerExtensions implements CloudEventExtension {
+
+		/**
+		 * Map storing the CloudEvent extensions extracted from message headers.
+		 */
+		private final Map<String, Object> cloudEventExtensions;
+
+		/**
+		 * Construct CloudEvent extensions by processing a message using expressions.
+		 *
+		 * @param message the Spring Integration message
+		 * @param expressions an array of {@link Expression}s where each accepts a message and returns a
+		 * {@code Map<String, Object>} of extensions
+		 */
+		@SuppressWarnings("unchecked")
+		ToCloudEventTransformerExtensions(EvaluationContext evaluationContext, Message<byte[]> message,
+				Expression @Nullable ... expressions) {
+			this.cloudEventExtensions = new HashMap<>();
```

Comment:
I prefer to have a blank line in the method body if header is multi-line.
Much easier to read the code.